tests/integration/admin_test.py

```diff
@@ -80,11 +84,161 @@ def test_update_groups(client: AtlanClient, group1: CreateGroupResponse):
     client.update_group(group)
 
 
-# TODO: users and then group membership
+def _get_user1_from_list(users: list[AtlanUser]) -> Optional[AtlanUser]:
+    for user in users:
+        if user.email == USER_EMAIL1.lower():
+            return user
+    return None
+
+
+def _get_user2_from_list(users: list[AtlanUser]) -> Optional[AtlanUser]:
+    for user in users:
+        if user.email == USER_EMAIL2.lower():
+            return user
+    return None
+
+
+def _get_user3_from_list(users: list[AtlanUser]) -> Optional[AtlanUser]:
+    for user in users:
+        if user.email == USER_EMAIL3.lower():
+            return user
+    return None
+
+
+@pytest.fixture(scope="module")
+def users(client: AtlanClient) -> Generator[list[AtlanUser], None, None]:
+    users: list[AtlanUser] = [
+        AtlanUser.create(email=USER_EMAIL1, role_name="$guest"),
+        AtlanUser.create(email=USER_EMAIL2, role_name="$guest"),
+        AtlanUser.create(email=USER_EMAIL3, role_name="$guest"),
+    ]
+    client.create_users(users=users)
+    created = client.get_users_by_email(email=EMAIL_DOMAIN)
+    assert created  # must be here for mypy
+    yield created
+    for user in created:
+        client.purge_user(str(user.id))
+
+
+@pytest.mark.order(after="test_retrieve_all_groups")
+def test_retrieve_users(client: AtlanClient, users: list[AtlanUser]):
+    global _default_group_count
+    all_users = client.get_all_users()
+    assert all_users
+    assert len(all_users) >= 3
+    users_list = client.get_users_by_email(USER_EMAIL1)
+    assert users_list
+    assert len(users_list) == 1
+    user1 = users_list[0]
+    assert user1
+    assert user1.id
+    assert user1.attributes
+    assert not user1.attributes.designation
+    assert user1.group_count == _default_group_count
+    users_list = client.get_users_by_email(EMAIL_DOMAIN)
+    assert users_list
+    assert len(users_list) == 3
+    assert _get_user1_from_list(users)
+    assert _get_user2_from_list(users)
+    assert _get_user3_from_list(users)
+
+
+@pytest.fixture(scope="module")
+def group2(
+    client: AtlanClient, users: list[AtlanUser]
+) -> Generator[CreateGroupResponse, None, None]:
+    to_create = AtlanGroup.create(GROUP_NAME2)
+    user2 = _get_user2_from_list(users)
+    assert user2  # must be here for mypy
+    user3 = _get_user3_from_list(users)
+    assert user3  # must be here for mypy
+    g = client.create_group(group=to_create, user_ids=[str(user2.id), str(user3.id)])
+    LOGGER.info(f"Created: {g}")
```

Comment:
Why are you logging the creation of the group? I think it just clutters the log file.

Reply:
Removed it